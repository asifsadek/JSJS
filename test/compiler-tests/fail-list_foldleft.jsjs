val msg = /\(s: string, b: bool): string => {
  if b then s else s ^ s;
};

// fail -> incorrect generic type resolution
val xs = List.fold_left(msg, true, ["hello", "world"]);
