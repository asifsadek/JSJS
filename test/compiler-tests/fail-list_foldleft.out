Type error: expected value of type 'bool', got a value of type 'string' instead

